Sources/NIOHTTP1/NIOTypedHTTPClientUpgradeHandler.swift

```diff
@@ -51,12 +51,12 @@ public struct NIOTypedHTTPClientUpgradeConfiguration<UpgradeResult: Sendable> {
 
     /// A closure that is run once it is determined that no protocol upgrade is happening. This can be used
     /// to configure handlers that expect HTTP.
-    public var notUpgradingCompletionHandler: @Sendable (Channel) -> EventLoopFuture<UpgradeResult>
+    public var notUpgradingCompletionHandler: @Sendable (Channel, HTTPResponseHead) -> EventLoopFuture<UpgradeResult>
```

Comment:
This isn't the exact shape of the change we'd make. To keep the API compatible, we'd need to have a new property with the new closure API, and then deprecate this one. We can then put them all into the same underlying private property, with just a reabstraction for the original one (which will drop the last argument).